refactor: make record validation filter content into a guide (#2446)
kroxylicious/kroxylicious: docs/_modules/record-validation/con-example-kafkaprotocolfilter-resource.adoc

// file included in the following:
//
// assembly-configuring-record-validation-filter.adoc

[id='con-example-kafkaprotocolfilter-resource-{context}']
= Example `KafkaProtocolFilter` resource

If your instance of Kroxylicious runs on Kubernetes, you must use a `KafkaProcotolFilter` resource to contain the filter configuration.

> Here's a complete example of a `KafkaProtocolFilter` resource configured for record encryption with Vault KMS:

Reviewer: ```suggestion
Here's a complete example of a `KafkaProtocolFilter` resource configured for record validation:
```

> .Example `KafkaProtocolFilter` resource

Reviewer: ```suggestion
.Example `KafkaProtocolFilter` resource for record validation
```

Reviewer: Is is really an example if it's got placeholders though? Maybe we should describe it as an outline or something

[source,yaml]
----
kind: KafkaProtocolFilter
metadata:

>   name: my-encryption-filter

Reviewer: ```suggestion
  name: my-validation-filter
```

spec:

>   type: RecordEncryption

Reviewer: ```suggestion
  type: RecordValidation
```
